IPython/utils/py3compat.py

```diff
@@ -27,310 +22,84 @@ def cast_unicode(s, encoding=None):
         return decode(s, encoding)
     return s
 
-def cast_bytes(s, encoding=None):
-    if not isinstance(s, bytes):
-        return encode(s, encoding)
-    return s
```

Comment:
I'm a bit wary of removing things from `py3compat` in case other modules like `ipykernel` rely on them.

Reply:
Sure, I'll make these changes. Seeing as Python 3 is now the only supported version, how about somehow marking `py3compat` as deprecated?

Reply:
We can mark it as deprecated once we are not using it anymore. If we mark it as deprecated and still use it here and there people complain we don't even follow our own deprecations.